### DISCUSSION TITLE
long term memory with a local qdrant vector db?

### DISCUSSION THREAD

**Discussion by ghost** [oobabooga/text-generation-webui#4065] [Ideas]
Would it make anyone happy to have a LTM module based on a vectordb such as qdrant? 

i never got superbooga to work so i built my own LTM code via docker-compose and a qdrant container:

* compose spins up a stock qdrant container locally, you have to specify a persistent volume somehow (.env/docker-compose.yml/etc)
* compose creates a vmnet to share packets between the qdrant container and ooba's container.
* the script.py sends each user input to the code, which automatically instantiates a qdrant collection based on character name unless one already exists.
* the user_input gets embeded with a SentenceTransformer and added to the vector db collection.  
* the user_input is used as a query to all previously embedded related comments
* adequately scoring comments are formatted and inserted into the prompt for relevant background access.

all config is still in the file.  maybe i could do a gradio gui to turn this on/off, allow for some tuning, delete and collection maintenance, etc. you could do stuff with this like upload a doc to the collection.

seems to me like allowing a bot access to everything it's ever heard before is a cool idea. the downside is i think a pull request would kind of clobber your docker strategy?

**Comment by olariuromeo** [comment]
nice , just instaled it
